src/common/root/index.js

```diff
@@ -47,9 +47,13 @@ export default function Root() {
       });
   }
 
+  const debouncedLint = useDebouncedCallback(() => {
+    lint();
+  }, 250);
+
   useEffect(() => {
-    debounce(lint, 250)();
-  });
+    debouncedLint();
+  }, [code, config, syntax]);
```

Comment:
[suggestion] How about using `useEffect` and `setTimeout` directly instead of using `useDebouncedCallback`?

Because `useDebouncedCallback` uses `useEffect` internally, I think more simple to understand the code behavior than nested `useEffect` by `useDebouncedCallback`.

```js
useEffect(() => {
  const timeoutID = setTimeout(lint, 250);
  return () => { clearTimeout(timeoutID) };
}, [code, config, syntax]);
```

Here is the `useDebouncedCallback` implementation: ⬇️ 
https://github.com/xnimorz/use-debounce/blob/91a7fe01d6b321a2bbe8b5c8e80f4242564b15c3/src/cache.js#L6-L14

Reply:
I think it's doesn't matter how any particular debounce function is implemented. We didn't copy `lodash.debounce` code, just to be able to see implementation.

We don't need to know implementations details. I think `useDebouncedCallback` is more clear than copy-pasted code because it's understandable right away what it is without understanding implementation.

Also, I could just copy useDebouncedCallback completely but decided not to do so, because in original repository this hook has tests, which is good.

Reply:
I misunderstood about `debounce`. Surely, it seems good to use the library than we reinvent `debounce`. 👍